aya/src/programs/cgroup_device.rs

```diff
@@ -0,0 +1,136 @@
+//! Cgroup device programs.
+use std::os::unix::prelude::{AsRawFd, RawFd};
+
+use crate::{
+    generated::{bpf_attach_type::BPF_CGROUP_DEVICE, bpf_prog_type::BPF_PROG_TYPE_CGROUP_DEVICE},
+    programs::{
+        define_link_wrapper, load_program, FdLink, Link, ProgAttachLink, ProgramData, ProgramError,
+    },
+    sys::{bpf_link_create, bpf_prog_attach, kernel_version},
+};
+
+/// A program used to watch or prevent device interaction from a cgroup.
+///
+/// [`CgroupDevice`] programs can be attached to a cgroup and will be called every
+/// time a process inside that cgroup tries to access (e.g. read, write, mknod)
+/// a device (identified through its major and minor number). See
+/// [mknod](https://man7.org/linux/man-pages/man2/mknod.2.html) as a starting point.
+///
+/// # Minimum kernel version
+///
+/// The minimum kernel version required to use this feature is [4.15](https://github.com/torvalds/linux/commit/ebc614f687369f9df99828572b1d85a7c2de3d92).
+///
+/// # Examples
+///
+/// ```no_run
+/// use aya::programs::CgroupDevice;
+///
+/// let cgroup = std::fs::File::open("/sys/fs/cgroup/unified")?;
+/// let program: &mut CgroupDevice = bpf.program_mut("cgroup_dev").unwrap().try_into()?;
```

Comment:
```suggestion
/// # let mut bpf = Bpf::load_file("ebpf_programs.o")?;
/// let program: &mut CgroupDevice = bpf.program_mut("cgroup_dev").unwrap().try_into()?;
```
`bpf` variable is missing

Reply:
Done, went ahead and drew from CgroupSysctl's docs!